tests/test_datasets/test_hierarchical_dataset.py

```diff
@@ -298,3 +446,80 @@ def test_to_hierarchical_dataset_hierarchical_structure(
+        ("long_hierarchical_structure", "long_df_level4", "D", "l2", "long_df_level2"),
+        ("long_hierarchical_structure", "long_df_level4", "D", "l1", "long_df_level1"),
+        ("long_hierarchical_structure", "long_df_level3", "D", "l2", "long_df_level2"),
+        ("long_hierarchical_structure", "long_df_level3", "D", "l1", "long_df_level1"),
+        ("long_hierarchical_structure", "long_df_level2", "D", "l1", "long_df_level1"),
+        ("tailed_hierarchical_structure", "tailed_df_level4", "D", "l3", "tailed_df_level3"),
+        ("tailed_hierarchical_structure", "tailed_df_level4", "D", "l2", "tailed_df_level2"),
+        ("tailed_hierarchical_structure", "tailed_df_level4", "D", "l1", "tailed_df_level1"),
+        ("tailed_hierarchical_structure", "tailed_df_level3", "D", "l2", "tailed_df_level2"),
+        ("tailed_hierarchical_structure", "tailed_df_level3", "D", "l1", "tailed_df_level1"),
+        ("tailed_hierarchical_structure", "tailed_df_level2", "D", "l1", "tailed_df_level1"),
+    ),
+)
+def test_get_level_dataset(hierarchical_structure_name, source_df_name, freq, target_level, target_df_name, request):
+    hierarchical_structure = request.getfixturevalue(hierarchical_structure_name)
+
+    source_df = request.getfixturevalue(source_df_name)
+    source_ts = TSDataset(df=source_df, freq=freq, hierarchical_structure=hierarchical_structure)
+
+    target_df = request.getfixturevalue(target_df_name)
+    target_ts = TSDataset(df=target_df, freq=freq, hierarchical_structure=hierarchical_structure)
+
+    estimated_target_ts = source_ts.get_level_dataset(target_level)
+
+    # check attributes
+    assert target_ts.freq == estimated_target_ts.freq
+    assert target_ts.hierarchical_structure == estimated_target_ts.hierarchical_structure
+    assert target_ts.current_df_level == estimated_target_ts.current_df_level
+
+    # check shape
+    assert target_ts.df.shape == estimated_target_ts.df.shape
+
+    # check column index
+    assert target_ts.df.columns.names == estimated_target_ts.df.columns.names
+    assert target_ts.df.columns.equals(estimated_target_ts.df.columns)
+
+    # check timeline index
+    assert target_ts.df.index.equals(estimated_target_ts.df.index)
+
+    # check values
+    assert target_ts.df.equals(estimated_target_ts.df)
+
+
+def test_get_level_dataset_invalid_level_name_error(market_level_df):
+    ts = TSDataset(df=market_level_df, freq="D")
+    with pytest.raises(ValueError, match="Method could be applied only to instances with a hierarchy!"):
+        ts.get_level_dataset(target_level="total")
+
+
+@pytest.mark.parametrize(
+    "target_level",
+    ("", "abcd"),
+)
+def test_get_level_dataset_invalid_level_name_error(simple_hierarchical_ts, target_level):
+    with pytest.raises(ValueError, match="Provided level name is not part of the hierarchy!"):
+        simple_hierarchical_ts.get_level_dataset(target_level=target_level)
+
+
+@pytest.mark.parametrize(
```

Comment:
May be just put the fixture names in the signature as there is only one parameters set